engine/src/main/kotlin/world/gregs/voidps/engine/client/update/batch/ZoneBatchUpdates.kt

```diff
@@ -49,10 +49,12 @@ class ZoneBatchUpdates : Runnable {
 
     fun run(player: Player) {
         val previousZone: Zone? = player["previous_zone"]
+        val previousLevel: Int? = player["previous_level"]
         val previous = previousZone?.toRectangle(radius = player.viewport!!.localRadius)?.toZones(player.tile.level)?.toSet()
```

Comment:
Why this works didn't make sense to me because `Zone` already includes a `level` so I had a look.
Turns out because `toZones(player.tile.level)` is using the current level, rather than the previous it means `previous` always contains the current zone unless teleporting.

So you can revert your changes and just change this to fix

```suggestion
        val previous = previousZone?.toRectangle(radius = player.viewport!!.localRadius)?.toZones(previousZone.level)?.toSet()
```

Reply:
Ahh, good catch. I'll make that change :)